rpmlint/readelfparser.py

```diff
@@ -0,0 +1,155 @@
+from itertools import dropwhile, takewhile
+import re
+import subprocess
+
+
+class ElfSection:
+    """
+    A simple wrapper representing one ELF section.
+    """
+    def __init__(self, name, size):
+        self.name = name
+        self.size = int(size, 16)
+
+
+class ElfProgramHeader:
+    """
+    A simple wrapper representing one ELF program header.
+    """
+    def __init__(self, name, flags):
+        self.name = name
+        self.flags = flags.replace(' ', '')
+        for flag in self.flags:
+            assert flag == 'R' or flag == 'W' or flag == 'E'
```

Comment:
This actually does not work as you would want. Asserts behave as in C so they get optimized out.

Reply:
Ah, that's new for me. I thought it's never optimized out:

```
test/test_binaries.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
rpmlint/checks/AbstractCheck.py:44: in check
    return self.check_binary(pkg)
rpmlint/checks/BinariesCheck.py:395: in check_binary
    self.run_all_checks(pkg, pkgfile.path, fname)
rpmlint/checks/BinariesCheck.py:290: in run_all_checks
    self.readelf_parser = ReadelfParser(pkgfile_path)
rpmlint/readelfparser.py:152: in __init__
    self.program_header_info = ElfProgramHeaderInfo(path)
rpmlint/readelfparser.py:116: in __init__
    self.parse()
rpmlint/readelfparser.py:139: in parse
    self.headers.append(ElfProgramHeader(r.group('header'), r.group('flags')))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rpmlint.readelfparser.ElfProgramHeader object at 0x7f4794ec75f8>, name = 'PHDR', flags = 'R E'

    def __init__(self, name, flags):
        self.name = name
        self.flags = flags.replace(' ', '')
        for flag in self.flags:
>           assert flag == 'R' or flag == 'W' or flag == 'e'
E           AssertionError
```